assets/src/scripts/monitoring-location/components/hydrograph/index.js

```diff
@@ -75,117 +66,80 @@ export const attachToNode = function(store,
         .select('.loading-indicator-container')
         .call(drawLoadingIndicator, {showLoadingIndicator: true, sizeClass: 'fa-3x'});
 
-    // Fetch time zone
-    const fetchTimeZonePromise = store.dispatch(timeZoneActions.retrieveIanaTimeZone(latitude, longitude));
-    // Fetch waterwatch flood levels
-    store.dispatch(floodDataActions.retrieveWaterwatchData(siteno));
-    let fetchDataPromise;
-    if (showOnlyGraph) {
-        // Only fetch what is needed
-        if (parameterCode && period) {
-            fetchDataPromise = store.dispatch(ivTimeSeriesDataActions.retrieveCustomTimePeriodIVTimeSeries(siteno, parameterCode, period));
-        } else if (parameterCode && startDT && endDT) {
-            // Don't fetch until time zone is available
-            fetchDataPromise = fetchTimeZonePromise.then(() => {
-                return store.dispatch(ivTimeSeriesDataActions.retrieveUserRequestedIVDataForDateRange(siteno, startDT, endDT, parameterCode));
-            });
+    const fetchDataPromise = store.dispatch(retrieveHydrographData(siteno, {
+        parameterCode: parameterCode,
+        period: startDT && endDT ? null : period || 'P7D',
+        startTime: DateTime.fromISO(startDT, {zone: config.locationTimeZone}),
+        endTime: DateTime.fromISO(endDT, {zone: config.locationTimeZone}),
+        loadCompare: compare,
+        loadMedian: false
+    }));
+
+    // if showing the controls, fetch the parameters
+    let fetchParameters;
+    if (!showOnlyGraph) {
+        fetchParameters = store.dispatch(retrieveHydrographParameters(siteno));
```

Comment:
The details of retrieveHydrographParameters are in the store section but it's worth mentioning here that the IV time series service (and gw service if applicable) is used to fetch the latest value (no period or startDT/endDT in the service call). So rather than 7 days of time series data for every parameter we only retrieve a single value. We only need to do this if the graph controls are rendered.